bfsgs/bfsgs.go

```diff
@@ -12,17 +13,25 @@ import (
 	giterator "google.golang.org/api/iterator"
 )
 
-type gsBucket struct {
-	bucket *storage.BucketHandle
-	config *Config
+func init() {
+	bfs.RegisterProtocol("gs", func(ctx context.Context, u *url.URL) (bfs.Bucket, error) {
+		q := u.Query()
+		return New(ctx, u.Host, &Config{
+			Prefix: q.Get("prefix"),
```

Comment:
Config is tiny right now, but what if it grows (gains new fields)? Still looks fine to me: don't like default impl - register yours. But I'd remove `panic` on double-register then (very minor).